tests/src/test/scala/com/nvidia/spark/rapids/RangeRetrySuite.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import com.nvidia.spark.rapids.Arm.withResource
+import com.nvidia.spark.rapids.jni.RmmSpark
+
+class RangeRetrySuite extends RmmSparkRetrySuiteBase {
+
+  test("Gpu range with split and retry OOM") {
```

Comment:
I think this test looks good, but I would like to see if you could add a test with other edge cases: 0 rows (start == end), negative step, 1 row, and also no retry as well.

Reply:
done